New state machine for Scheduler and Elevator subsystems, completed Scheduler algorithm for Iteration 3

pathumd opened the pull request:
Things completed in this branch:
- Scheduler and Elevator's state machine has been redone to allow Iteration 3 functionality
- Scheduler algorithm designed and implemented to allow Iteration 3 functionality
- General code improvements

NOTE: Please take a read through the 'Scheduler algorithm' section of the README to get an understanding of how it works.

yehandesilva committed:
Floor system small fixes

yehandesilva committed:
Removed synchronizer from floor subsytem

yehandesilva committed:
Floor request constructor change

yehandesilva committed:
Elevator status completion

yehandesilva committed:
Elevator status now contains elevator id

yehandesilva committed:
Floor subsystem differentiates between floor requests and elevator status

yehandesilva committed:
Elevator receiver fix

pathumd committed:
Implemented new state diagram for Scheduler

- Implemented new state diagram for Scheduler to support Iteration 3 functionality
- Reworked Scheduler to now act as the context class
- Added concrete state classes (WaitingForPacket, CheckingPacketType, ProcessingElevatorStatus, and ProcessingFloorRequest)
- Implementation of entry/do/exit actions for ProcessingElevatorStatus and ProcessingFloorRequest are yet to be completed

yehandesilva committed:
Elevator State Machine Change

yehandesilva committed:
Merge branch 'iteration-3-scheduler-algorithm' into iteration-3-elevator-state-refactoring

yehandesilva committed:
Merge pull request #27 from Aranesh22/iteration-3-elevator-state-refactoring

Elevator State Machine Change

yehandesilva committed:
ElevatorTaskQueue class

yehandesilva committed:
Removed synchronizer from Scheduler constructor

yehandesilva committed:
Cleanup

yehandesilva committed:
Floor subsystem to use send receive socket

yehandesilva committed:
Elevator state machine fix

yehandesilva committed:
Elevator run method

yehandesilva committed:
ElevatorTaskQueue elevator status getter

yehandesilva committed:
Elevator state machine changes

yehandesilva committed:
ElevatorTaskQueue addFloorRequest method

yehandesilva committed:
ElevatorTaskQueue IP address addition

yehandesilva committed:
ElevatorReceiver sends initial msg

pathumd committed:
Implemented ProcessingElevatorStatus and ProcessingFloorRequest states

- Implemented ProcessingElevatorStatus and ProcessingFloorRequest states
- Made additional changes to Scheduler

yehandesilva committed:
Various bugfixes

yehandesilva committed:
Elevator context hashmap

yehandesilva committed:
Elevator state change print change

yehandesilva committed:
Scheduler state transition fix

yehandesilva committed:
Moved Scheduler state files into a singular file

yehandesilva committed:
Removed synchronizer class

yehandesilva committed:
Elevator state machine change

yehandesilva committed:
Elevator method re-ordering

yehandesilva committed:
Open/close doors methods

yehandesilva committed:
Elevator state machine final fixes

yehandesilva committed:
Better floor requests to test system

yehandesilva committed:
Added doorsOpened field to elevator

yehandesilva committed:
DoorsOpened field added to elevator status class

yehandesilva committed:
Elevator load time and travel time fields

pathumd committed:
Updated implementation of Scheduler algorithm

- Updated implementation of Scheduler algorithm to account for FloorRequests that cannot be served immediately

yehandesilva committed:
Initial elevator setup fix

yehandesilva committed:
toUdpStringBytes fix

pathumd committed:
FloorRequest being served removed from list

yehandesilva committed:
Prevent elevator from opening/closing doors multiple times

yehandesilva committed:
Only remove floor once it has been visited

yehandesilva committed:
Added Elevator State Diagram

yehandesilva committed:
Algorithm suitable elevators change

Elevator is not suitable if elevator is moving away from floor request even if floor request and elevator direction are the same.

pathumd committed:
Updated README to include explanation of Scheduler's algorithm

- Updated README to include details on the Scheduler's algorithm, as well as a brief communication overview of the system

harishan-a reviewed (approved):
Looks good. Read README.

harishan-a committed:
State Machine for Scheduler

Lindiks committed:
Testing - In progress

Currently working on Testing - investigating how to test the more nested thread heavy classes such as Elevator, scheduler, etc.. will also look into network testing for the future

Lindiks committed:
Testing - In progress

Organized the tests and created the final files to work on
